### PR TITLE
tests, integration-tests: port change errors test to spread

### PR DESCRIPTION
**Pull request by fgimenez**

### CODE REVIEW

**Comment by mvo5** [comment]
:+1:

**Comment by mvo5** [comment]
The diff looks good, there is a error in the spread tests though that looks unreleated:
```
Content-Length: 64

{"type":"sync","status-code":200,"status":"OK","result":["TBD"]}+ snap install --channel=beta ubuntu-core

65.56 MB / 65.56 MB  100.00 % 16.16 MB/s 

error: cannot list snaps: cannot communicate with server: Get http://localhost/v2/snaps: read 
```
I wonder what is going on there, I had seen (rare) events like this before, wonder if the daemon crashes/panics sometimes? would be nice to get the syslog output on failed tests.

**Comment by niemeyer** [comment]
Travis is nuts.. the build is passing now:

image

File: tests/main/change-errors/task.yaml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+summary: Checks for cli errors of the change command.
+
+execute: |
+  echo "When an invalid ID is given to the change command it shows an error"
+  if snap change 10000000; then
+    echo "Expected error when trying change on invalid ID" && exit 1
```
Comment (niemeyer) [inline tests/main/change-errors/task.yaml]: The `&& exit 1` here would nicer as `\nexit 1`.. trivial and not a blocker, though. Merging..

**Comment by fgimenez** [comment]
> I wonder what is going on there, I had seen (rare) events like this before, wonder if the daemon crashes/panics sometimes? would be nice to get the syslog output on failed tests.

Yes, this was due to reset problems, it's fixed in master with #1424 

@niemeyer @mvo5 thanks! :)

**Pull request merged**